src/widgets/SparkWebuiMainAreaWidget.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { VDomRenderer } from '@jupyterlab/apputils';
+import { JupyterFrontEnd } from '@jupyterlab/application';
+import { launchIcon } from '@jupyterlab/ui-components';
+import { JupyterLabAppContext } from '../const';
+import { ServerConnection } from '@jupyterlab/services';
+import { URLExt } from '@jupyterlab/coreutils';
+
+export interface SparkWebuiWidgetOptions {
+  app: JupyterFrontEnd;
+}
+
+const WIDGET_CLASS = 'jp-SparkConnectExtensionSparkWebuiWidget';
+
+export class SparkWebuiMainAreaWidget extends VDomRenderer {
+  app: JupyterFrontEnd;
+  requestUrl: string;
+
+  constructor(options: SparkWebuiWidgetOptions) {
+    super();
+    super.addClass(WIDGET_CLASS);
+
+    const { app } = options;
+    this.app = app;
+
+    const settings = ServerConnection.makeSettings();
+    this.requestUrl = URLExt.join(
+      settings.baseUrl,
+      'spark-connect-labextension', // API Namespace
```

Comment:
You have this hardcoded here and bellow, when you were using a variable in other places. I would stick to a variable everywhere.

Reply:
Whoops minor oversight, sorry.